### ISSUE TITLE
Error in porous MATLAB code

### ISSUE THREAD

**Issue by mountaindust** [mountaindust/IB2d#2]
@nickabattista

It looks like there is a bug in please_Compute_Porous_Slip_Velocity.m: the divide there should be element-wise: ./, not /. 

With just a regular / (as is), you end up getting something rather weird for Up_X and Up_Y - a matrix of all zeros except one column in the middle which has the values you want. I don't know why this happens, but I saw it in the debugger. These two matrices of mostly zeros then get put together to form Por_Mat, which in turn is given back to IBM_Driver. IBM_Driver then expects a two column matrix full of useful values so calls on the first two columns - but will just get all zeros in return. As a result, xLag and yLag never are updated, and the porous system has no effect.

Obviously, this can be fixed by making the division operator element-wise, but the reason I am raising the issue here (and tagging you on it, Nick), is that when I fix this bug by adding back in the element-wise operation it breaks the example and the simulation dies due to some NaN generation. This is how I found the problem in the first place - I ported it to Python element-wise and was trying to figure out why my code was breaking, then tracked the bug to this part of the MATLAB version.

Can you figure out why the code is breaking once this fix is inserted? I would just skip porting porous media for now, but I think Laura has an upcoming project she wants to do with porous stuff, so it would be good to get this working.

**Comment by nickabattista** [comment]
Hi Christopher,

Finally sat down and looked at this in greater depth. I fixed the bug, but
have not committed or pushed the corrected version yet. It appears that
once I added the actual component-wise division (e.g., the bug that you
caught - thanks, again!), and lowered the porosity parameter a few orders
of magnitude, it seems to work like a gem. I want to do further testing,
though. Thanks, again!!

Congrats on the Savanna paper, again!

Cheers,
Nick
